apps/condo/domains/resident/tasks/helpers/notifyOnPaydayisCurrnetDateValid.js

```diff
@@ -0,0 +1,13 @@
+const dayjs = require('dayjs')
+
+function notifyOnPaydayIsCurrentDateValid (today) {
+    const targetDayOfSendingThePush = dayjs().set('date', 20).day()
+
+    // https://day.js.org/docs/en/get-set/day
+    if ((targetDayOfSendingThePush === 0 || targetDayOfSendingThePush === 6) && ( today.format('DD') > 16 && today.day() === 5)) {
```

Comment:
Got it.
"If target day is a weekday, remind on Friday before it"
Drop it to the comments.